sentinel/lib/info-utils.js

```diff
@@ -51,29 +57,41 @@ module.exports = {
           if (err) {
             return reject(err);
           }
-          resolve();
+          return resolve(infoDoc);
         };
-
         if (err) {
           return reject(err);
         }
-
         if (infoDoc) {
           infoDoc.latest_replication_date = new Date();
-          return db.medic.insert(infoDoc, done);
+          if(!infoDoc.transitions) {
+            infoDoc.transitions = change.doc.transitions || {};
+          }
+          return db.sentinel.insert(infoDoc, done);
```

Comment:
If I'm reading this correctly the infoDoc will always be inserted even if it's already completely up to date. Perhaps we only need to do the insert if `!infoDoc.transitions` (ie: in the block one line above)?